packages/core/test/util/runtime.test.ts

```diff
@@ -514,6 +531,96 @@ test('evalEnablement disable invalid case based on schema condition', t => {
     ruleValue: 'bar'
   };
   t.is(evalEnablement(uischema, data, undefined, createAjv()), false);
-  t.is(evalEnablement(uischema, { ...data, ruleValue: 'baz' }, undefined, createAjv()), false);
-  t.is(evalEnablement(uischema, { ...data, ruleValue: 'foo' }, undefined, createAjv()), true);
+  t.is(
+    evalEnablement(
+      uischema,
+      { ...data, ruleValue: 'baz' },
+      undefined,
+      createAjv()
+    ),
+    false
+  );
+  t.is(
+    evalEnablement(
+      uischema,
+      { ...data, ruleValue: 'foo' },
+      undefined,
+      createAjv()
+    ),
+    true
+  );
+});
+
+test('isInherentlyEnabled disabled globally', t => {
+  t.false(
+    isInherentlyEnabled(
+      { jsonforms: { readonly: true } },
+      null,
+      null,
+      null,
+      null,
+      null
+    )
+  );
+});
+
+test('isInherentlyEnabled disabled by ownProps', t => {
+  t.false(
+    isInherentlyEnabled(null, { enabled: false }, null, null, null, null)
+  );
+});
+
+test('isInherentlyEnabled enabled by ownProps', t => {
+  t.true(isInherentlyEnabled(null, { enabled: true }, null, null, null, null));
+});
+
+test('isInherentlyEnabled disabled by uischema', t => {
+  t.false(
+    isInherentlyEnabled(
+      null,
+      null,
+      ({ options: { readonly: true } } as unknown) as ControlElement,
+      null,
+      null,
+      null
+    )
+  );
+});
+
+test('isInherentlyEnabled disabled by schema', t => {
+  t.false(
+    isInherentlyEnabled(null, null, null, { readonly: true }, null, null)
+  );
+});
+
+test('isInherentlyEnabled disabled by rule', t => {
+  const leafCondition: LeafCondition = {
+    type: 'LEAF',
+    scope: '#/properties/ruleValue',
+    expectedValue: 'bar'
+  };
+  const uischema: ControlElement = {
+    type: 'Control',
+    scope: '#/properties/value',
+    rule: {
+      effect: RuleEffect.DISABLE,
+      condition: leafCondition
+    }
+  };
+  const data = {
+    value: 'foo',
+    ruleValue: 'bar'
+  };
+
+  t.false(isInherentlyEnabled(null, null, uischema, null, data, null));
+});
+
+test('isInherentlyEnabled disabled by config', t => {
+  t.false(
+    isInherentlyEnabled(null, null, null, null, null, { readonly: true })
+  );
+});
+
+test('isInherentlyEnabled enabled', t => {
+  t.true(isInherentlyEnabled(null, null, null, null, null, null));
 });
```

Comment:
Would be great to add some tests which also test the combination of multiple readOnlys being present, especially when they overwrite each other. For example when an `ownProp.enabled` is `false` but a `uischema.option.readonly: false` is set.

Obviously testing all combinations is a bit much but a reasonable subset would be great.